lib/icingadb/icingadb.hpp

```diff
@@ -27,6 +31,186 @@
 namespace icinga
 {
 
+/**
+ * RedisKey is the enumeration of all Redis keys used by IcingaDB.
+ *
+ * Each enum value represents a specific Redis key type from which the actual Redis key strings are derived.
+ * For instance, the `Host` enum value corresponds to the Redis key pattern `icinga:host`. These enums help
+ * in organizing and managing the various Redis keys in a transparent and consistent manner and avoid hardcoding
+ * key strings throughout the codebase.
+ *
+ * @ingroup icingadb
+ */
+enum class RedisKey : uint8_t
+{
+	/* Command-related keys */
+	CheckCmdArg,
+	CheckCmdEnvVar,
+	CheckCmdCustomVar,
+
+	EventCmdArg,
+	EventCmdEnvVar,
+	EventCmdCustomVar,
+
+	NotificationCmdArg,
+	NotificationCmdEnvVar,
+	NotificationCmdCustomVar,
+
+	/* Dependency related config */
+	DependencyNode,
+	DependencyEdge,
+	RedundancyGroup,
+
+	/* Hosts & Services */
+	HostCustomVar,
+	HostGroupMember,
+	HostGroupCustomVar,
+
+	ServiceCustomVar,
+	ServiceGroupMember,
+	ServiceGroupCustomVar,
+
+	/* Users & Usergroups */
+	UserCustomVar,
+	UserGroupMember,
+	UserGroupCustomVar,
+
+	/* Notification */
+	NotificationUser,
+	NotificationUserGroup,
+	NotificationRecipient,
+	NotificationCustomVar,
+
+	/* Timeperiods */
+	TimePeriodRange,
+	TimePeriodInclude,
+	TimePeriodExclude,
+	TimePeriodCustomVar,
+
+	/* Downtimes */
+	ScheduledDowntimeCustomVar,
+
+	/* State keys marker and state entries */
+	_state_keys_begin,
+	HostState,
+	ServiceState,
+	RedundancyGroupState,
+	DependencyEdgeState,
+	_state_keys_end,
+};
+
+/**
+ * Dirty bits for config/state changes.
+ *
+ * These are used to mark objects as "dirty" in order to trigger appropriate updates in Redis.
+ * Each bit represents a different type of change that requires a specific action to be taken.
+ *
+ * @ingroup icingadb
+ */
+enum DirtyBits : uint
```

Comment:
I don't know how you've discovered `uint`, but it's not a standard C++ type and in my system headers it's defined in a block with the comment `/* Old compatibility names for C types.  */`, so that doesn't really sound like something that should be used.

Reply:
It states `/* Sys V compatibility */` on my machine :).